code/functions.py

```diff
@@ -0,0 +1,165 @@
+import numpy as np
+import holoviews as hv
+from holoviews import plotting
+from holoviews import Options, Store
+import pfaffian as pf
```

Comment:
I don't agree, because the function called in `pf` is called `pfaffian`, so usage: `pf.pfaffian`.